test/integration/test_cluster.c

```diff
@@ -97,17 +97,16 @@ TEST(cluster, restart, setUp, tearDown, 0, cluster_params)
 	struct rows rows;
 	long n_records =
 	    strtol(munit_parameters_get(params, "num_records"), NULL, 0);
-	char sql[128];
 
 	HANDSHAKE;
 	OPEN;
 	PREPARE("CREATE TABLE test (n INT)", &stmt_id);
 	EXEC(stmt_id, &last_insert_id, &rows_affected);
 
+	PREPARE("INSERT INTO TEST(n) VALUES(?)", &stmt_id);
 	for (int i = 0; i < n_records; ++i) {
-		sprintf(sql, "INSERT INTO test(n) VALUES(%d)", i + 1);
-		PREPARE(sql, &stmt_id);
-		EXEC(stmt_id, &last_insert_id, &rows_affected);
+		EXEC_PARAMS(stmt_id, &last_insert_id, &rows_affected,
+			    {.type = SQLITE_INTEGER, .integer = i});
```

Comment:
A fun fact is that this macro invocation sees *five* arguments, the last two of which are `{.type = SQLITE_INTEGER` and `.integer = i}`.